docs/src/Services/Reference/ReflectionHelper.php

```diff
@@ -0,0 +1,365 @@
+                    return $this->outputFormatter->linkClasses($namedType);
+                }, $type->getTypes());
+
+                $typedParameters[] = implode('|', $namedTypes).' '.$this->outputFormatter->addCssClasses($parameterName, ['token', 'variable']).$this->getDefaultValueString($parameter);
+            }
+            if ($type instanceof \ReflectionIntersectionType) {
+                $namedTypes = array_map(function (\ReflectionNamedType $namedType) {
+                    return $this->outputFormatter->linkClasses($namedType);
+                }, $type->getTypes());
+
+                $typedParameters[] = implode('&', $namedTypes).' '.$this->outputFormatter->addCssClasses($parameterName, ['token', 'variable']).$this->getDefaultValueString($parameter);
+            }
+            if ($type instanceof \ReflectionNamedType) {
+                $typedParameters[] = $this->outputFormatter->linkClasses($type).' '.$this->outputFormatter->addCssClasses($parameterName, ['token', 'variable']).$this->getDefaultValueString($parameter);
+            }
+        }
+
+        return $typedParameters;
+    }
+
+    private function getDefaultValueString(\ReflectionParameter $parameter): string
+    {
+        if (!$parameter->isDefaultValueAvailable()) {
+            return '';
+        }
+
+        return match ($parameter->getDefaultValue()) {
+            null => ' = null',
+            [] => ' = []',
+            default => ' = '.$parameter->getDefaultValue()
+        };
+    }
+
+    private function getParameterName(\ReflectionParameter $parameter): string
+    {
+        return $parameter->isPassedByReference() ? '&$'.$parameter->getName() : '$'.$parameter->getName();
+    }
+
+    private function getReturnType(\ReflectionMethod $method): string
+    {
+        $type = $method->getReturnType();
+
+        if (!$type) {
+            return '';
+        }
+
+        if ($type instanceof \ReflectionUnionType) {
+            return implode('|', array_map(function (\ReflectionNamedType $reflectionNamedType): string {
+                return $this->outputFormatter->linkClasses($reflectionNamedType);
+            }, $type->getTypes()
+            ));
+        }
+        if ($type instanceof \ReflectionIntersectionType) {
+            return implode('&', array_map(function (\ReflectionNamedType $reflectionNamedType): string {
+                return $this->outputFormatter->linkClasses($reflectionNamedType);
+            }, $type->getTypes()
+            ));
+        }
+
+        return $this->outputFormatter->linkClasses($type);
+    }
+
+    public function containsOnlyPrivateMethods(\ReflectionClass $reflectionClass): bool
+    {
+        // Do not skip empty interfaces
+        if (interface_exists($reflectionClass->getName()) || trait_exists($reflectionClass->getName())) {
+            return false;
+        }
+
+        if (!empty($reflectionClass->getProperties())) {
```

Comment:
```suggestion
        if (!$reflectionClass->getProperties()) {
```

not sure empty is needed